packages/sdk/src/client/index.ts

```
@@ -195,28 +232,30 @@ export class WunderGraphClient<Role> {
 		})();
 	};
 
-	private subscribeWithFetch = <Input, Data>(
-		subscription: SubscriptionProps<Partial<InternalSubscriptionArgsWithInput<Input>>>,
+	private subscribeWithFetch = <
+		OperationName extends keyof Operations['subscriptions'],
+		Input extends Operations['subscriptions'][OperationName]['input'] = Operations['subscriptions'][OperationName]['input'],
+		Data = Operations['subscriptions'][OperationName]['data'],
+		LiveQuery extends Operations['subscriptions'][OperationName]['isLiveQuery'] = Operations['subscriptions'][OperationName]['isLiveQuery']
+	>(
+		subscription: SubscriptionProps<OperationName, InternalClientSubscriptionArgs<Input>, LiveQuery>,
 		cb: (result: SubscriptionResult<Data>) => void
 	) => {
 		(async () => {
 			try {
 				const params = this.queryString({
-					wg_variables: subscription?.input,
+					wg_variables: this.hasInput(subscription) ? subscription.input : {},
 					wg_live: subscription?.isLiveQuery ? true : undefined,
 				});
-				const response = await this.fetch(
-					this.baseURL + '/' + this.applicationPath + '/operations/' + subscription.operationName + params,
-					{
-						headers: {
-							...this.extraHeaders,
-							'Content-Type': 'application/json',
-							'WG-SDK-Version': this.sdkVersion,
-						},
-						method: 'GET',
-						signal: subscription?.abortSignal,
-					}
-				);
+				const response = await this.fetch(this.operationUrl(String(subscription.operationName), params), {
+					headers: {
+						...this.extraHeaders,
+						'Content-Type': 'application/json',
+						'WG-SDK-Version': this.sdkVersion,
+					},
+					method: 'GET',
+					signal: subscription?.abortSignal,
+				});
 
 				if (!this.isOK(response) || response.body == null) {
 					cb(this.errorResult(response));
```

This currently swallows the original error. https://github.com/wundergraph/wundergraph/blob/main/packages/sdk/src/client/index.ts#L314

The should be able to access the original error.

---

This is a bit more complicated. `errorResult` is fine because it only executes for requests with statusCode != `200` so we are talking about gateway errors. The issue with this implementation is that the client maps statusCodes to gateway errors. We should just pass the gateway error to the client.

---

This issue existed already in the client. I would say we create another issue to improve this, so we can ship the improved types.

---

ok